zebra-chain/src/sapling/keys.rs

```diff
@@ -690,10 +695,11 @@ impl From<(AuthorizingKey, NullifierDerivingKey)> for IncomingViewingKey {
 impl FromStr for IncomingViewingKey {
     type Err = SerializationError;
 
+    #[allow(clippy::unwrap_in_result)]
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         match bech32::decode(s) {
             Ok((hrp, bytes, Variant::Bech32)) => {
-                let decoded = Vec::<u8>::from_base32(&bytes).unwrap();
+                let decoded = Vec::<u8>::from_base32(&bytes).expect("bytes are valid base32");
```

Comment:
This code parses untrusted data, so the reason needs to explain why the code will never panic

```suggestion
                let decoded = Vec::<u8>::from_base32(&bytes).expect("bech32::decode guarantees valid base32");
```